kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/PurchaseOrderRetransmitDocument.java

```diff
@@ -0,0 +1,204 @@
+     * transmit date is updated.
+     *
+     * @see org.kuali.kfs.module.purap.document.PurchaseOrderDocument#doRouteStatusChange()
+     */
+    @Override
+    public void doRouteStatusChange(DocumentRouteStatusChange statusChangeEvent) {
+        super.doRouteStatusChange(statusChangeEvent);
+
+        try {
+            // DOCUMENT PROCESSED
+            if (this.getFinancialSystemDocumentHeader().getWorkflowDocument().isProcessed()) {
+                SpringContext.getBean(PurchaseOrderService.class).setCurrentAndPendingIndicatorsForApprovedPODocuments(this);
+                setPurchaseOrderLastTransmitTimestamp(SpringContext.getBean(DateTimeService.class).getCurrentTimestamp());
+                updateAndSaveAppDocStatus(PurchaseOrderStatuses.APPDOC_OPEN);
+            }
+            // DOCUMENT DISAPPROVED
+            else if (this.getFinancialSystemDocumentHeader().getWorkflowDocument().isDisapproved()) {
+                SpringContext.getBean(PurchaseOrderService.class).setCurrentAndPendingIndicatorsForDisapprovedChangePODocuments(this);
+
+                // for app doc status
+                try {
+                    String nodeName = SpringContext.getBean(WorkflowDocumentService.class).getCurrentRouteLevelName(this.getFinancialSystemDocumentHeader().getWorkflowDocument());
+                    String reqStatus = PurapConstants.PurchaseOrderStatuses.getPurchaseOrderAppDocDisapproveStatuses().get(nodeName);
+                    updateAndSaveAppDocStatus(PurapConstants.PurchaseOrderStatuses.getPurchaseOrderAppDocDisapproveStatuses().get(reqStatus));
+                } catch (WorkflowException e) {
+                    logAndThrowRuntimeException("Error saving routing data while saving App Doc Status " + getDocumentNumber(), e);
+                }
+            }
+            // DOCUMENT CANCELED
+            else if (this.getFinancialSystemDocumentHeader().getWorkflowDocument().isCanceled()) {
+                SpringContext.getBean(PurchaseOrderService.class).setCurrentAndPendingIndicatorsForCancelledChangePODocuments(this);
+                // for app doc status
+                updateAndSaveAppDocStatus(PurapConstants.PurchaseOrderStatuses.APPDOC_CANCELLED);
+            }
+        }
+        catch (WorkflowException e) {
+            logAndThrowRuntimeException("Error saving routing data while saving document with id " + getDocumentNumber(), e);
+        }
+    }
+
+    public boolean isShouldDisplayRetransmitTab() {
+        return shouldDisplayRetransmitTab;
+    }
+
+    public void setShouldDisplayRetransmitTab(boolean shouldDisplayRetransmitTab) {
+        this.shouldDisplayRetransmitTab = shouldDisplayRetransmitTab;
+    }
+
+    @Override
+    protected boolean shouldAdhocFyi() {
+        return false;
+    }
+    
+    /**
+     * Gets the routeCodeObj attribute. 
+     * @return Returns the routeCodeObj.
+     */
+    public Building getBuildingObj() {
```

Comment:
Please add:
    if( ObjectUtils.isNull(buildingObj) ){
            this.refreshReferenceObject("buildingObj");
     }